framework/src/meshgenerators/ConcentricCircleMeshGenerator.C

```diff
@@ -983,5 +983,6 @@ ConcentricCircleMeshGenerator::generate()
   LaplaceMeshSmoother lms(*mesh);
   lms.smooth(_smoothing_max_it);
 
+  mesh->set_isnt_prepared();
```

Comment:
to be consistent with the other MGs that create a mesh from scratch, you should prepare_for_use here

Reply:
I thought that we had come together as a group and decided that we wanted to remove calls to `prepare_for_use` in favor of `set_isnt_prepared`

Reply:
With the predicate that we have a method for determining whether the mesh is actually prepared ... which we now have with `valid_is_prepared`

Reply:
Just saying there are lots of small ones that still prepare for use at the end, and I don’t think there s any harm in that

Reply:
What I care most strongly about is being consistent across mesh generators, whether they are creating the mesh from scratch or taking an input. I think being concerned about following mesh generators not checking the prepared status when the prepared status matters for that generator is valid. But I also think there are valid performance concerns of calling `prepare_for_use` for every generator (although it would be nice to have some profiling that shows what the most expensive parts of `prepare_for_use` are). 

That being said, I thought we had come to a consensus on `set_isnt_prepared`. I don't want to keep debating it. If you are a strong advocate for `prepare_for_use`, then let's have another discussion with @roystgnr, @friedmud, and @loganharbour.

Reply:
> What I care most strongly about is being consistent across mesh generators

```
(moose) giudgl@fn654105:~/projects/moose_v2/framework/src/meshgenerators$ grep -r prepare_for_use
./PatchMeshGenerator.C:  mesh->prepare_for_use();
./RinglebMeshGenerator.C:  mesh->prepare_for_use();
./ConcentricCircleMeshGenerator.C:    mesh->prepare_for_use();
./FillBetweenSidesetsGenerator.C:    mesh->prepare_for_use();
./DistributedRectilinearMeshGenerator.C:  mesh->prepare_for_use();
./PolyLineMeshGenerator.C:  mesh.prepare_for_use();
./AnnularMeshGenerator.C:  mesh->prepare_for_use();
./SpiralAnnularMeshGenerator.C:  mesh->prepare_for_use();
./LowerDBlockFromSidesetGenerator.C:    mesh->prepare_for_use();
./FileMeshGenerator.C:    mesh->prepare_for_use();
```

Consistency here, currently at least, means preparing for use after creating something (small usually).

> I also think there are valid performance concerns of calling prepare_for_use for every generator

I agree with that, but small generators do not have performance concerns for prepare_for_use, unless in some extremely narrow cases

> If you are a strong advocate for prepare_for_use

I wouldnt describe myself as such haha. I just think that several mesh generators still don't know that they need a prepared mesh, and it's only working in most cases right now because of additional/cautious prepare_for_use calls. 

This was a suggestion btw. I dont recall if I "requested changes" but I dont think I did

Reply:
You’re advising opposite of what we decided as a group. But you make a good point. I will go change the generators you listed to be consistent with our group decision

Reply:
Thinking about this more, I think I could be amenable to having a division in treatment where initial mesh gens are generally prepared and follow on mesh gens are generally marked with isn’t prepared. This would support the case where you want to delete your remote elements before extruding. And this would also help mesh gens like the lower d block which use the partitioning of interior parent elements to determine lower d element partitioning. 

Then if someone brings profiles showing poor performance due to an initial mesh gen prepare, we could consider switching that mesh gen to just setting isn’t prepared.

Reply:
But then again maybe the generators I just mentioned should be calling the prepares at the top of generate instead since they’re the ones who need it (goes back to @friedmud’s argument).  And the initial mesh doesn’t need to guess whether it would be good for follow on gens to have the mesh prepared

Reply:
It's a trade off between being safe (preparing too much) and hoping we have caught all the MGs that need a prepare before they act (one day we will). I m happy with any point in between, considerable effort has already been spent on this